blog: upstream-led contributions and team practices (#412)
2i2c-org/2i2c-brochure: content/blog/2025/good-citizen/index.md

---

> title: "The difference between upstream-led vs. stakeholder-led open source contributions"

Reviewer: I would like to spend some time brainstorming different options here. I'm not a big fan of this one, although I agree that the earlier 'strategic vs tacitical' is not appropriate either.

Author: For what it's worth I'm not attached to this language at all. Happy to use something else. The reason I moved away from the previous language was (so you know my rationale in case it's helpful for next iteration):

- it wasn't precise enough (and I don't want people to fill in too many missing gaps)
- All of our work needs to be strategic at the end of the day, and in our case getting our product work accomplished through open source development is definitely a strategic choice

Reviewer: 100% i never liked the previous ones as well for both the reasons you listed, plus the military / violent feel of 'tactical' in my mouth

3. [Adding landing pages functionality to Jupyter Book and MyST](https://github.com/jupyter-book/myst-theme/pull/531) was work we did to support member communities like [CryoCloud](https://cryointhecloud.org) and [Project Pythia](https://cookbooks.projectpythia.org/).

The fact that these are open source contributions is *incidental*. We are making open source contributions, but we are _primarily_ doing this work to deliver value to _our community network_. 

### How 2i2c wants to plan team work around stakeholder-led contributions

Stakeholder-led contributions naturally align with 2i2c's overall goals and strategy, so we can re-use our pre-existing product processes for planning and delivering on them. However, we also want to provide transparency to upstream communities so that they understand who is driving the contributions that we're making.

With that in mind, here are a few ways that stakeholder-led contributions relate to our practices:

- Stakeholder-led contributions should be defined by our product roadmap and prioritization processes.

>   - For engineering, this process eventually feeds into the Engineering Project Board under the `product-dev` (from product board), `internal-eng` (from recurring calendar events and internal eng roadmap), `tech-support` (via freshdesk) and `partnership-support`  (via slack) buckets.

Reviewer: ```suggestion
```
We don't do this anymore

---

> title: "Our approach to providing directed and foundational contributions to open source communities"

Author: ```suggestion
title: "How organizations can support a healthy open source ecosystem through directed and foundational contributions"
```

> title: "How organizations can support a healthy open source ecosystem through directed and foundational contributions"

Author: ```suggestion
title: "On being a good open source citizen: supporting a healthy ecosystem through directed and foundational contributions"
```
